### ISSUE TITLE
Current errors when compiling on Windows

### ISSUE THREAD

**Issue by Croydon** [inexor-game/code#92]
When one is trying to compile Inexor with Visual Studio 2013 on Windows 7 following errors appear:
(sorry for that bad german translation, Visual Studio don't allow me for some reasons to install the language pack..)

>Fehler	21	error C2668: 'min': Mehrdeutiger Aufruf einer überladenen Funktion	\src\shared\tools.h	257	1	server
Fehler	22	error C2660: 'memcpy': Funktion akzeptiert keine 2 Argumente	src\shared\tools.h	257	1	server
Fehler	23	error C2668: 'min': Mehrdeutiger Aufruf einer überladenen Funktion	src\shared\tools.h	258	1	server
Fehler	49	error C2668: 'min': Mehrdeutiger Aufruf einer überladenen Funktion	src\shared\tools.h	257	1	inexor
Fehler	50	error C2660: 'memcpy': Funktion akzeptiert keine 2 Argumente	shared\tools.h	257	1	inexor
Fehler	51	error C2668: 'min': Mehrdeutiger Aufruf einer überladenen Funktion	src\shared\tools.h	258	1	inexor

Also a ton of warnings, but I suppose the most aren't that important, but maybe this one:
>Warnung	1	warning google/protobuf/compiler/parser.cc: 471] No syntax specified for the proto file. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)	C:\**\build\src\CUSTOMBUILD	inexor_rpc_protobuf

Moreover, when I'm generating the Visual Studio project files I got following from cmake:
>CMake Warning (dev) at src/CMakeLists.txt:54 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  >Quoted variables like "MSVC" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
This warning is for project developers.  Use -Wno-dev to suppress it.

Same attempt to compile it with Code::Blocks 13.12 on Windows 7 with GNU GCC version 4.7.1 32-bit occurs following errors AND also the very same errors with GNU GCC 4.9.2 64-bit (however I don't get any warnings here, maybe a configuration thing? Also it's not in the offical Code::Blocks release download package bundled with the IDE at the time of writing):

>/src/platform_windows/lib32/libprotobuf.a(descriptor.o): In function `ZNKSs8_M_checkEjPKc':
/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)'
/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)'
/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)'
/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)'
/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)'
/src/platform_windows/lib32/libprotobuf.a(descriptor.o):/usr/i686-w64-mingw32/include/c++/4.9.2/bits/basic_string.h:324: more undefined references to `std::__throw_out_of_range_fmt(char const*, ...)' follow
||=== Build failed: 6 error(s), 86 warning(s) (1 minute(s), 33 second(s)) ===|

**Comment by a-teammate** [comment]
```
Warning	1	warning C4541: 'dynamic_cast' used on polymorphic type 'google::protobuf::Message' with /GR-; unpredictable behavior may result (inexor\code\src\rpc\inexor_service.pb.cc)	inexor\code\src\platform_windows\include\google\protobuf\stubs\casts.h	91	1	inexor
Warning	2	warning C4996: 'std::_Copy_impl': Function call with parameters that may be unsafe - this call relies on the caller to check that the passed values are correct. To disable this warning, use -D_SCL_SECURE_NO_WARNINGS. See documentation on how to use Visual C++ 'Checked Iterators' (inexor\code\src\rpc\inexor_service.pb.cc)	D:\Program Files (x86)Microsoft Visual Studio 12.0\VC\include\xutility	2132	1	inexor
Error	3	error C2668: 'min' : ambiguous call to overloaded function (inexor\code\src\rpc\InexorServiceImpl.cpp)	inexor\code\src\shared\tools.h	257	1	inexor
Error	4	error C2660: 'memcpy' : function does not take 2 arguments (inexor\code\src\rpc\InexorServiceImpl.cpp)	inexor\code\src\shared\tools.h	257	1	inexor
Error	5	error C2668: 'min' : ambiguous call to overloaded function (inexor\code\src\rpc\InexorServiceImpl.cpp)	inexor\code\src\shared\tools.h	258	1	inexor
	6	IntelliSense: more than one instance of overloaded function "min" matches the argument list:
            function template "T min(T a, T b)"
            function template "const _Ty &std::min(const _Ty &_Left, const _Ty &_Right)"
            argument types are: (unsigned long long, size_t)	inexor\code\src\shared\tools.h	173	19	inexor
	7	IntelliSense: more than one instance of overloaded function "max" matches the argument list:
            function template "T max(T a, T b)"
            function template "const _Ty &std::max(const _Ty &_Left, const _Ty &_Right)"
            argument types are: (int, int)	inexor\code\src\shared\tools.h	320	61	inexor
	8	IntelliSense: more than one instance of overloaded function "max" matches the argument list:
            function template "T max(T a, T b)"
            function template "const _Ty &std::max(const _Ty &_Left, const _Ty &_Right)"
            argument types are: (int, int)	inexor\code\src\shared\tools.h	1032	34	inexor
	9	IntelliSense: "string" is ambiguous	inexor\code\src\shared\tools.h	1302	96	inexor
```

**Comment by a-teammate** [comment]
[libprotobuf WARNING google/protobuf/compiler/parser.cc:471] No syntax specified for the proto file. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)

**Comment by aschaeffer** [comment]
Replace min by std::min ?

**Comment by a-teammate** [comment]
yeah i tried that and didnt work.
turned out the way i tried it was the important part. 
so thats fixed now.
Still remaining: tons of linking errors.

**Comment by a-teammate** [comment]
fixed with the merge of cmake-update #89
